modules/account-lib/src/coin/sol/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,91 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BuildTransactionError } from '../baseCoin/errors';
+import { Transaction } from './transaction';
+import { isValidAmount, validateAddress } from './utils';
+import { TransactionType } from '../baseCoin';
+import { InstructionBuilderTypes } from './constants';
+import { TokenTransfer } from './iface';
+
+import assert from 'assert';
+import { TransactionBuilder } from './transactionBuilder';
+import { PublicKey } from '@solana/web3.js';
+import { ASSOCIATED_TOKEN_PROGRAM_ID, Token, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+
+export interface SendParams {
+  address: string;
+  amount: string;
+  mintAddress: string;
```

Comment:
I feel like we might have gotten this backwards - maybe we should have this be `tokenName` instead of mint address.
then we do something like `coins.from(tokenName)` to check if the token is supported or not.

then get token specific details (`mintAddress`, `decimals`, etc.)

OK to leave this as is since we're doing the same for creating ATAs. will think about this some more

Reply:
Good one I like that - either store fields we need in the Transfer object or store whole coin on the transaction object